probe_src/probe_py/analysis.py

```diff
@@ -1,44 +1,46 @@
 import typing
-import networkx
+import networkx as nx
 from .parse_probe_log import ProvLog, Op, CloneOp, ExecOp, WaitOp, CLONE_THREAD
 
+# [pid, exec_epoch_no, tid, Op | str, op_index]
+Node: typing.TypeAlias = tuple[int, int, int, Op | str, int]
+program_order_edges = list[tuple[Node, Node]]()
+fork_join_edges = list[tuple[Node, Node]]()
+exec_edges = list[tuple[Node, Node]]()
 
-def construct_process_graph(process_tree_prov_log: ProvLog) -> str:
-    """
-    Construct a happens-before graph from process_tree_prov_log
-
-    The graph shows the order that prov ops happen.
-    """
-    Node: typing.TypeAlias = tuple[int, int, int, Op | str]
+def provlog_to_digraph(process_tree_prov_log: ProvLog) -> nx.digraph:
```

Comment:
I think both instances of `nx.digraph` need to be capitalized like `nx.DiGraph`. I currently do not implement type-checking in Python, so those annotations are currently ignored, but it is on my list to start doing type-checking.